### PR TITLE
Add support for min aggregation

### PR DESCRIPTION
**Pull request by Mitep**

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/package.scala [hunk 1]
```diff
@@ -67,6 +60,18 @@ package object elasticsearch extends IndexNameNewtype with RoutingNewtype {
...
       zio.flatMap(_.aggregations)
 
+    /**
+     * Executes the [[ElasticRequest.SearchRequest]] or the [[ElasticRequest.SearchAndAggregateRequest]].
+     *
+     * @param name
+     *   the name of the aggregation to retrieve
+     * @return
+     *   a [[RIO]] effect that, when executed, will produce the aggregation as instance of
+     *   [[result.AvgAggregationResult]].
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/package.scala:70]: Check if this compiles without warnings.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala [hunk 2]
```diff
@@ -141,6 +141,19 @@ object ElasticAggregationSpec extends ZIOSpecDefault {
             equalTo(Max(name = "aggregation", field = "intField", missing = Some(20.0)))
           )
         },
+        test("min") {
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala:144]: Can you add aggregationWithAllParams? For encoding too.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala (hunk 2 above)
Comment (Mitep) [reply modules/library/src/test/scala/zio/elasticsearch/ElasticAggregationSpec.scala:144]: Test already includes case classes where we cover all fields.

File: modules/library/src/main/scala/zio/elasticsearch/package.scala (hunk 1 above)
Comment (Mitep) [reply modules/library/src/main/scala/zio/elasticsearch/package.scala:70]: I don't see any warning regarding RIO, only a couple regarding unused methods.

**Review by dbulaja98 — APPROVED** [review]

File: docs/overview/aggregations/elastic_aggregation_min.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+val aggregation: MinAggregation = minAggregation(name = "minAggregation", field = "intField")
+```
+
+You can create a [type-safe](https://lambdaworks.github.io/zio-elasticsearch/overview/overview_zio_prelude_schema) `Min` aggregation using the `minAggregation` method this way:
+```scala
+// Document.intField must be number value, because of Min aggregation
+val aggregation: MinAggregation = minAggregation(name = "minAggregation", field = Document.intField)
+```
+
+If you want to change the `missing`, you can use `missing` method:
```
Comment (markaya) [inline docs/overview/aggregations/elastic_aggregation_min.md:25]:
```suggestion
If you want to change the `missing` parameter, you can use `missing` method:
```

**Review by markaya — CHANGES_REQUESTED** [review]

**Review by markaya — APPROVED** [review]

**Pull request merged**